crypto/threads_pthread.c

```diff
@@ -638,8 +791,42 @@ __owur int CRYPTO_THREAD_read_lock(CRYPTO_RWLOCK *lock)
 __owur int CRYPTO_THREAD_write_lock(CRYPTO_RWLOCK *lock)
 {
 # ifdef USE_RWLOCK
+#  ifdef REPORT_RWLOCK_CONTENTION
+    struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);
+
```

Comment:
would it make sense to do `contention_fp == NULL` check here? it does not make sense to collect data when there is no consumer (file).